xds/src/main/java/io/grpc/xds/LookasideLb.java

```diff
@@ -147,17 +150,26 @@ public int getPriority() {
         }
 
         /**
-         * A synthetic policy name identified by balancerName. The implementation detail doesn't
+         * A synthetic policy name identified by xds config. The implementation detail doesn't
          * matter.
          */
         @Override
         public String getPolicyName() {
-          return "xds_child_policy_balancer_name_" + newBalancerName;
+          return "xds_policy__balancer_name_" + newXdsConfig.balancerName
```

Comment:
Although you said it's your implementation detail. I am dare to say this doesn't look elegant...

Reply:
What about
```
MoreObjects.toStringHelper(XdsConfig.class)
          .add("balancerName", balancerName)
          .add("childPolicy", childPolicy)
          .add("fallbackPolicy", fallbackPolicy)
          .add("edsServiceName", edsServiceName)
          .add("lrsServerName", lrsServerName)
          .toString();
```

Reply:
That's not appropriate, it produces an object representation string, which is used to represent the instance of this class. Obviously, that's not equivalent to policy name. Maybe leave the current setting if we can't come up with a better one. My feeling is that, if we found many things for a class awkward or inappropriate, probably the design or idea for this class is inappropriate.

Reply:
I agree `getPolicyName()` is awkward. This is only awkward thing I find for using `GracefulSwitchLoadBalancer`. (Other thing like difficulty to test is not big issue, it's more difficult to test without it.)

Reply:
As offline discussed, changing `GracefulSwitchLoadBalancer` API to switch based on equality of `LoadBalancerProvider` instead of based on policy name will make this awkwardness go away. Will do it in separate PR as this requires API change of another class.